pkg/operators/v1/operatorgroup_types.go

```diff
@@ -108,35 +108,36 @@ func (o *OperatorGroup) HasServiceAccountSynced() bool {
 	return false
 }
 
-// GetLabel returns a label that is applied to Namespaces to signify that the
-// namespace is a part of the OperatorGroup using selectors.
-func (o *OperatorGroup) GetLabel() string {
-	key, _ := o.OGLabelKeyAndValue()
-	return key
-}
-
 // OGLabelKeyAndValue returns a key and value that should be applied to namespaces listed in the OperatorGroup.
-func (o *OperatorGroup) OGLabelKeyAndValue() (string, string) {
-	return fmt.Sprintf(OperatorGroupLabelTemplate, o.GetNamespace(), o.GetName()), ""
+// If the UID is not set an error is returned.
+func (o *OperatorGroup) OGLabelKeyAndValue() (string, string, error) {
+	if string(o.UID) == "" {
+		return "", "", fmt.Errorf("Missing UID")
+	}
+	return fmt.Sprintf(OperatorGroupLabelTemplate, string(o.UID)), "", nil
```

Comment:
maybe use `GetUID()` so that we can get rid of the `error` return here. Since the previous implementation also considers not getting name/ns is not an error to be handled here. It is invalid for a server to return an object that does not have UID.
Wonder if that makes sense, else just a comment about using GetUID(). thanks!

Reply:
nvm, since it was @benluddy's idea to add it.

Reply:
We should probably still use `GetUID()` since it's the appropriate getter.

Reply:
Updated.